src/OpenTelemetry/Metrics/Exemplar/SimpleExemplarReservoir.cs

```diff
@@ -57,9 +57,13 @@ public override Exemplar[] Collect(ReadOnlyTagCollection actualTags, bool reset)
             if (reset)
             {
                 this.runningExemplars[i].Timestamp = default;
-                this.measurementsSeen = 0;
             }
-        }
+        }
+
+        // Reset internal state irrespective of temporality.
+        // This ensures incoming measurements have fair chance
+        // of making it to the reservoir.
+        this.measurementsSeen = 0;
```

Comment:
Trying to fix line endings via github too 🤞...

```suggestion
        }

        // Reset internal state irrespective of temporality.
        // This ensures incoming measurements have fair chance
        // of making it to the reservoir.
        this.measurementsSeen = 0;
```

Reply:
hehe. didn't work. @cijothomas fix your line endings 😄

Reply:
oops. I used a temporary cloud box to code this, and that machine has something weird about line endings! Let me fix it,